### PR TITLE
feat: improve fuel injection

### PR DESCRIPTION
**Pull request by luizstacio**
This PR improves how we detect when fuel is injected on the page. With this PR now, it's possible to listen for an event on the document object `FuelLoaded` this name makes reference to `DOMContentLoaded`. This event is triggered right after the fuel object is assigned to the target object, in this case, the window.

This PR also adds a new handler to `onInstall` event. The new handler injects `fuel` into the tabs already open, removing the need to refresh the page.

These two improvements enable applications to listen to the event `FuelLoaded`, and once the user installs the Wallet, the application can start work without the need for workarounds or request for the user to refresh the page.

close: #308

### CODE REVIEW

File: packages/docs/docs/how-to-use.mdx [hunk 1]
```diff
@@ -14,6 +14,15 @@ window.fuel.connect();
 
 You can try this code directly in the developer console.
 
+## Detect when fuel is loaded
+
+It's possible that your applaction loads before `window.fuel` is injected. To detect when the `fuel` is loaded and ready to use, you can listen to the event `FuelLoaded` on the document.
```
Comment (tomiiide) [inline packages/docs/docs/how-to-use.mdx:19]:
```suggestion
It's possible that your application loads before `window.fuel` is injected. To detect when the `fuel` is loaded and ready to use, you can listen to the event `FuelLoaded` on the document.
```

**Review by tomiiide — CHANGES_REQUESTED** [review]
Looks good, just one little typo.

**Review by tomiiide — APPROVED** [review]
Looks good now.

File: packages/docs/examples/FuelLoaded.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+    // If fuel is already loaded, call the handler
+    if (window.fuel) {
+      onFuelLoaded();
+    }
+
+    // Listen for the fuelLoaded event
+    document.addEventListener('FuelLoaded', onFuelLoaded);
+
+    // On unmount, remove the event listener and clear the timeout
```
Comment (eswarasai) [inline packages/docs/examples/FuelLoaded.tsx:23]:
```suggestion
    // On unmount, remove the event listener
```

**Review by LuizAsFight — APPROVED** [review]

**Review by LuizAsFight — CHANGES_REQUESTED** [review]
tests breaking after merging with master

**Pull request merged**